pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -835,6 +835,13 @@ func (m *kubeGenericRuntimeManager) GetPodStatus(uid kubetypes.UID, name, namesp
 	})
 	glog.V(4).Infof("getSandboxIDByPodUID got sandbox IDs %q for pod %q", podSandboxIDs, podFullName)
 
+	// Get statuses of all containers visible in the pod.
```

Comment:
@yujuhong Well, makes sense. Just found the sandbox uid label well suited for this case, I've reverted the last commit during rebase, and add a new method for getting podIP through containers filtered by sandboxID.

For sandbox on windows, it is now created only, but not in use. I think it's related to the current network support implementation on windows.

Docker on windows recently supports shared pod network mode: https://github.com/moby/moby/pull/30657, we could have a try to see whether can make use of it to support sandbox.